.github/workflows/ci.yml

```
@@ -45,23 +54,48 @@ jobs:
         run: |
           sudo apt update
           sudo apt install libpulse-dev libdbus-1-dev libncursesw5-dev libxcb-shape0-dev libxcb-xfixes0-dev
-      - uses: actions/checkout@v2
-        name: Checkout src
-      - uses: actions/cache@v2
-        with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-      - name: Running cargo check
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --locked --release --target ${{ matrix.target }} ${{ matrix.features }}
+      - name: Running cargo build
+        run: cargo build --locked --target ${{ matrix.target }} ${{ matrix.features }}
 
+  # Check Rust code formatting.
   fmt:
+    name: Running `cargo fmt`
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          target: x86_64-unknown-linux-gnu
+          components: clippy, rustfmt
       - name: cargo fmt
         run: cargo fmt --all -- --check
+
+  # Run `cargo clippy` on all the targets in all workspace members with all
+  # features enabled, and return an error if there are any clippy suggestions.
+  clippy:
+    name: Running `cargo clippy`
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        name: Cache build data
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            ~/.cargo/bin
+            target
```

```suggestion
            ~/.cargo/bin/
            ~/.cargo/registry/index/
            ~/.cargo/registry/cache/
            ~/.cargo/git/db/
            target/
```
I guess these would also need to be adjusted (if we go for it).